public/application.js

```diff
@@ -12,3 +11,32 @@ function loadMoreRepositories(event) {
   }
   xhr.send();
 }
+
+function toggleStar(el) {
+  fetch(el.href, { method: 'PUT' }).then(function(response) {
```

Comment:
I went with `fetch` here because I figure people looking at this example will have modern browsers.